### PR TITLE
fix deps (GH migration)

### PR DESCRIPTION
**Pull request by bap2pecs**
## Summary

- companion PR for https://github.com/babylonlabs-io/babylon-private/pull/1
- fixed `github.com/babylonchain/finality-provider` to `github.com/babylonlabs-io/finality-provider`
- merged commits from `dev`

## Test Plan
```
make lint
make test
make test-e2e-op
make test-e2e-babylon
```

### CODE REVIEW

**Comment by SebastianElvis** [comment]
Thanks for doing this!

Maybe we can take this chance to merge all commits from `dev` as well?

**Comment by bap2pecs** [comment]
> Maybe we can take this chance to merge all commits from `dev` as well?

sure but that can probably be in a separate PR. I am still running tests for this PR and if nothing goes wrong, I expect we can merge this today

File: itest/opstackl2/op_test_manager.go [hunk 1]
```diff
@@ -728,7 +728,7 @@ func (ctm *OpL2ConsumerTestManager) SetupFinalityProviders(
 	stakingParams []stakingParam,
 ) []*service.FinalityProviderInstance {
 	// A BTC delegation has to stake to at least one Babylon finality provider
-	// https://github.com/babylonchain/babylon-private/blob/3d8f190c9b0c0795f6546806e3b8582de716cd60/x/btcstaking/keeper/msg_server.go#L220
+	// https://github.com/babylonlabs-io/babylon-private/blob/3d8f190c9b0c0795f6546806e3b8582de716cd60/x/btcstaking/keeper/msg_server.go#L220
```
Comment (bap2pecs) [inline itest/opstackl2/op_test_manager.go:731]: this is incorrect replacement. will fix it

Comment (bap2pecs) [reply itest/opstackl2/op_test_manager.go]: fixed

**Comment by bap2pecs** [comment]
all other tests are passing except `make test-e2e-wasmd` and `make test-e2e-bcd`

this is what I got:
```
=== RUN   TestConsumerFpDataInjection
    babylon_node_handler.go:153: babylon log file: /var/folders/9_/q4wsdnh14_s60_74cd2rbztm0000gp/T/zBabylonTest343434105/babylon.log
Starting babylond with command:  /Users/zidong/go/bin/babylond start --home=/var/folders/9_/q4wsdnh14_s60_74cd2rbztm0000gp/T/zBabylonTest343434105/node0/babylond --log_level=trace --trace
service injective.evm.v1beta1.Msg does not have cosmos.msg.v1.service proto annotation
    wasmd_handler.go:172:
        	Error Trace:	/Users/zidong/Documents/Projects/babylon-finality-provider/itest/cosmwasm/wasmd/wasmd_handler.go:172
        	            				/Users/zidong/Documents/Projects/babylon-finality-provider/itest/cosmwasm/wasmd/wasmd_handler.go:42
        	            				/Users/zidong/Documents/Projects/babylon-finality-provider/itest/cosmwasm/wasmd/wasmd_test_manager.go:72
        	            				/Users/zidong/Documents/Projects/babylon-finality-provider/itest/cosmwasm/wasmd/wasmd_e2e_test.go:28
        	Error:      	Received unexpected error:
        	            	error running command: exit status 2
        	Test:       	TestConsumerFpDataInjection
--- FAIL: TestConsumerFpDataInjection (2.77s)
```

**Comment by SebastianElvis** [comment]
The base branch in `babylon-private` is sync'ed with the `dev` branch, and there were some breaking changes. So I guess we have to do renaming and merge in a single PR. Probably we need to merge `dev` branch first, then address conflicts, and finally fiox all tests

**Comment by bap2pecs** [comment]
@SebastianElvis I just finished testing everything. the only failing tests are `make test-e2e-wasmd` and `make test-e2e-bcd`. all other tests are passing.

**Comment by bap2pecs** [comment]
> Maybe we can take this chance to merge all commits from `dev` as well?

done

**Comment by bap2pecs** [comment]
tested again. still all tests passed except test-e2e-wasmd and test-e2e-bcd

I also verifier that they are failing at `base/consumer-chain-support`. so it seems it's not caused by this PR.

File: tools/go.mod [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
 go 1.21
 
 toolchain go1.21.4
 
 require (
 	github.com/CosmWasm/wasmd v0.51.0
-	github.com/babylonchain/babylon v0.9.0-rc.2
 	github.com/babylonchain/babylon-sdk/demo v0.0.0-20240705194516-4e2c5650cde8
+	github.com/babylonlabs-io/babylon v0.9.0-rc.2
```
Comment (SebastianElvis) [inline tools/go.mod:10]: Now we have v0.9.0 and maybe we can use that

**Review by SebastianElvis — APPROVED** [review]
Great work!

File: Dockerfile [hunk 3]
```diff
@@ -17,12 +17,12 @@ RUN git config --global url."git@github.com:".insteadOf "https://github.com/"
 ENV GOPRIVATE=github.com/babylonchain/*
```
Comment (SebastianElvis) [inline Dockerfile:17]:
```suggestion
ENV GOPRIVATE=github.com/babylonlabs-io/*
```

**Pull request merged**